### PR TITLE
feature(journal): human-readable UUIDs via reference mapping

### LINKED ISSUE #629: Store Human Readable references in `debtor/creditor` tables
In order to facilitate easy lookups of the debtor/creditor ids in the vouchers and journal tables, I propose that we store the human readable ids in the `debtor/creditor` tables.  The only change is add a single text column `reference` that stores `TPA1`, `TPA304`, etc, as the debtor or creditor's human-readable ID.

### LINKED ISSUE #630: Namespace human readable references by entity type
The human-readable IDs of 1.x were highly useful for distinguishing entities of the same type, but did not carry much information to separate entities of different types.  For example, an employee might have an employee reference `TPA3` , while a different patient might also have a patient reference `TPA3`.

I propose that we add a single letter prefix to the reference, the entity type.  Some examples are shown below:
1. `patient` - `PTPA2445`, `PTPA3234`.  The leading `P` is for patient.
2. `employee` - `ETPA1`, `ETPB345`.  The leading `E` is for an employee.
3. `supplier` - `CTPA343`, `CTPC999`.  The leading `C` is for creditor/supplier.
4. `cash` - ? `ZTPA1245`, `ZTPA5`.  The leading `Z` is for the cash payment.  If someone has a better idea, please comment!
5. `invoice` - `ITPA31`, `IPTC8`.  The leading `I` is for invoice.
6. `voucher` - `VTPA2`, `VTPB72`. The leading `V` is for voucher.

Since we control the rendering of the human readable ID, we can make these any format we desire to try and make it more readable.  The proposal above is the standard proposal.  However, other formats might be:
- `V.TPA.2` - separation with periods
- `I-TPB-941` - separation with dashes

### LINKED ISSUE #753: (proposal) Move MySQL triggers into separate triggers.sql file
To better navigate our codebase, it would be helpful to move all our triggers out into a separate `triggers.sql` file similar to the `procedures.sql` file.

### PR DESCRIPTION
**Pull request by jniles**
This PR introduces human readable UUIDs to the posting journal by introducing reference mapping tables into the database.  These tables are updated whenever a record with a UUID is added via MySQL triggers.

Two tables are introduced:
1. `entity_map` which maps `entity_uuid`s to readable text
2. `document_map` which maps `record_uuid`s and `reference_uuid`s to readable text.

This creates a much nicer accounting user experience, since the records in the posting journal are suddenly meaningful.  See the image below to understand what the pull request does:

journalreadableuuids
_Fig 1: Human Readable IDs_

Closes #629.  Closes #630.  Closes #753.

---

Thank you for contributing!

Before submitting this pull request, please verify that you have:
- [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
- [x] Run integration tests.
- [x] Run end-to-end tests.
- [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

**Comment by jniles** [comment]
@mbayopanda can I get a review?

File: client/src/js/services/JournalService.js [hunk 1]
```diff
@@ -2,25 +2,13 @@ angular.module('bhima.services')
...
-  service.read = read;
-
-  /** Getting posting journal data */
-   function read() {
-     return $http.get(baseUrl)
-      .then(util.unwrapHttpResponse);
-   }
+function JournalService(Api) {
+  var service = new Api('/journal/');
+  return service;
```
Comment (mbayopanda) [inline client/src/js/services/JournalService.js:13]: 👍

File: server/controllers/admin/employees.js [hunk 2]
```diff
@@ -162,14 +162,7 @@ function lookupEmployee(id) {
...
 
-  return db.exec(sql, [id])
-  .then(function (rows) {
-    if (rows.length === 0) {
-      throw new NotFound(`Could not find an employee with id ${id}.`);
-    }
-
-    return rows[0];
-  });
+  return db.one(sql, [id], id, 'employee');
```
Comment (mbayopanda) [inline server/controllers/admin/employees.js:165]: 👍

File: server/controllers/finance/journal/index.js [hunk 3]
```diff
@@ -120,6 +118,11 @@ function getTransaction(req, res, next) {
  * POST /journal/:uuid/reverse
  */
 function reverse(req, res, next) {
+<<<<<<< 3fa08c9a78e043908319ddaad1dc050257f02b0d
+=======
+  const uid = db.bid(req.params.uuid);
+  const userId = req.session.user.id;
+>>>>>>> feat(entities): introduce entity_map
```
Comment (mbayopanda) [inline server/controllers/finance/journal/index.js]: Is this normal ?

File: server/controllers/finance/journal/index.js [hunk 4]
```diff
@@ -119,11 +120,6 @@ function getTransaction (req, res, next){
  * POST /journal/:uuid/reverse
  */
 function reverse(req, res, next) {
-<<<<<<< 3fa08c9a78e043908319ddaad1dc050257f02b0d
-=======
-  const uid = db.bid(req.params.uuid);
-  const userId = req.session.user.id;
->>>>>>> feat(entities): introduce entity_map
```
Comment (mbayopanda) [inline server/controllers/finance/journal/index.js]: 👍

**Review by mbayopanda — COMMENTED** [review]
It's LGTM, you have just to update your branch

**Comment by jniles** [comment]
Will do.  Thanks for the review!

File: server/controllers/finance/journal/index.js (hunk 4 above)
Comment (jniles) [reply server/controllers/finance/journal/index.js]: :(

Comment (jniles) [reply server/controllers/finance/journal/index.js]: Ah, I didn't realize this was from such an old commit.  That's what happens when you rebase your branch too many times ...

**Comment by jniles** [comment]
Merging as passed tests and review.

**Pull request merged**